src/libostree/ostree-sysroot.c

```diff
@@ -813,6 +815,17 @@ ostree_sysroot_load_if_changed (OstreeSysroot  *self,
                                cancellable, error))
     return FALSE;
 
+  /* Determine whether we're "physical" or not, the first time we initialize */
+  if (!self->loaded)
+    {
+      if (fstatat (self->sysroot_fd, "sysroot", &stbuf, 0) < 0)
```

Comment:
I'd say we are also testing for "not physical" too... if `/sysroot` exists that's a strong signal it's not physical.

Checking the inodes is interesting.  Hmm...maybe a simple one is to check whether we're in the booted deployment to mean not-physical. That would cover the 98% case.